ebu_tt_live/bindings/validation.py

```diff
@@ -159,43 +194,155 @@ def _post_timing_set_attribute(self, attr_use):
+                proposed_end = min(dataset['timing_syncbase'] + end_timedelta, dataset['timing_end_stack'][-1])
+            # New end
+            else:
+                proposed_end = dataset['timing_syncbase'] + end_timedelta
+
+        if proposed_end is not None:
+            dataset['timing_end_stack'].append(proposed_end)
+            dataset['timing_end_limit'] = max(dataset.get('timing_end_limt', timedelta()), proposed_end)
+            self._computed_end_time = proposed_end
+
+        # Store the element's activation begin times
+        if begin_timedelta is not None:
+            # Let's push it onto the stack. These assignments must happen last otherwise the syncbase will be wrong
+            # in calculations happening after syncbase adjustment.
+            dataset['timing_begin_stack'].append(begin_timedelta)
+            dataset['timing_syncbase'] += begin_timedelta
+
+        # If we have a non-zero availability time we need to factor it in BUT the syncbase stays
+        if dataset['availability_time']:
+            self._computed_begin_time = max(dataset['timing_syncbase'], dataset['availability_time'])
+        else:
+            self._computed_begin_time = dataset['timing_syncbase']
+
+        if self._computed_begin_time is not None:
+            if dataset['timing_begin_limit'] is not None and dataset['timing_begin_limit'] < self._computed_begin_time or dataset['timing_begin_limit'] is None:
+                # This means that timing begin limit needs updating
+                dataset['timing_begin_limit'] = self._computed_begin_time
+
+
+    def _semantic_postprocess_timing(self, dataset, element_content):
+        begin_timedelta = None
+        end_timedelta = None
+
+        if hasattr(self, 'end') and self.end is not None or hasattr(self, 'dur') and self.dur is not None:
+            # We pushed on the stack it is time to pop it
+            end_timedelta = dataset['timing_end_stack'].pop()
+
+        if hasattr(self, 'begin') and self.begin is not None:
+            # We pushed on the stack it is time to pop it
+            begin_timedelta = dataset['timing_begin_stack'].pop()
+            dataset['timing_syncbase'] -= begin_timedelta
+
+        # If the forward running part of the traversal could not assign an end time we can use the backwards route
+        # which is in a way similar to dynamic programming because we take the children computed times and take the
+        # maximum value from them. SPECIAL case: a single leaf element with an undefined end time renders the entire
+        # branch undefined
+
+        if end_timedelta is not None and self.computed_end_time is None \
+                or end_timedelta is None and self.computed_end_time is not None:
+            # This is just a simple sanity check. It should never be triggered.
+            # Should the calculation be changed this filters out an obvious source of error.
+            raise LogicError()
```

Comment:
And so I wonder if the best solution is would not be to simply add :

```
self._computed_begin_time = None
self._computed_end_time = None
```
At the beginning of the `_semantic_preprocess_timing` method.
As everything is computed again when we run the validation the second time, it does not create problems, at least no problems I can see and it solves our `LogicError()` problem.

@kozmaz87 please tell me what you think about that